src/client/src/orangehrmDashboardPlugin/components/EmployeesOnLeaveWidget.vue

```diff
@@ -0,0 +1,142 @@
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <base-widget
+    icon="leavelist"
+    icon-type="svg"
+    class="emp-leave-chart"
+    :empty="isEmpty"
+    :empty-text="emptyText"
+    :loading="isLoading"
+    :title="$t('general.employees_on_leave_today')"
+  >
+    <div v-for="leave in leaveList" :key="leave" class="orangehrm-leave-card">
+      <div class="orangehrm-leave-card-profile-image">
+        <img
+          alt="profile picture"
+          class="employee-image"
+          :src="`../pim/viewPhoto/empNumber/${leave.empNumber}`"
+        />
+      </div>
+      <div class="orangehrm-leave-card-details">
+        <oxd-text tag="p" class="orangehrm-leave-card-emp-name">
+          {{ leave.empName }}
+        </oxd-text>
+        <oxd-text
+          v-if="leave.leaveType"
+          tag="p"
+          class="orangehrm-leave-card-leave-details"
+        >
+          {{ leave.leaveType }}
+        </oxd-text>
+      </div>
+      <oxd-text tag="p" class="orangehrm-leave-card-emp-id">
+        {{ leave.employeeId }}
+      </oxd-text>
+    </div>
+  </base-widget>
+</template>
+
+<script>
+import {APIService} from '@/core/util/services/api.service';
+import {freshDate, formatDate} from '@ohrm/core/util/helper/datefns';
+import BaseWidget from '@/orangehrmDashboardPlugin/components/BaseWidget.vue';
+import useEmployeeNameTranslate from '@/core/util/composable/useEmployeeNameTranslate';
+
+export default {
+  name: 'EmployeesOnLeaveWidget',
+
+  components: {
+    'base-widget': BaseWidget,
+  },
+
+  setup() {
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      'api/v2/dashboard/employees/leaves',
+    );
+    const {$tEmpName} = useEmployeeNameTranslate();
+
+    return {
+      http,
+      tEmpName: $tEmpName,
+    };
+  },
+
+  data() {
+    return {
+      leaveList: [],
+      leavePeriod: null,
+      isLoading: false,
+    };
+  },
+
+  computed: {
+    isEmpty() {
+      return this.leaveList.length === 0;
+    },
+    emptyText() {
+      return this.leavePeriod
+        ? this.$t('general.no_employees_are_on_leave_today')
+        : this.$t('general.leave_period_not_defined');
+    },
+  },
+
+  beforeMount() {
+    this.isLoading = true;
+    this.http
+      .getAll({
+        date: formatDate(freshDate(), 'yyyy-MM-dd'),
+      })
+      .then(response => {
+        const {data, meta} = response.data;
+        this.leaveList = data.map(item => {
+          const {employee, leaveType, duration} = item;
+          let _leaveType = leaveType?.name;
+          if (_leaveType && duration === 'half_day_morning') {
+            _leaveType += ` (${this.$t('leave.half_day_morning')})`;
+          }
+          if (_leaveType && duration === 'half_day_afternoon') {
+            _leaveType += ` (${this.$t('leave.half_day_evening')})`;
+          }
+          if (_leaveType && duration === 'specify_time') {
+            _leaveType += ` (${item.startTime} - ${item.endTime})`;
+          }
```

Comment:
just for my understanding? do we have composable for this in the leave module?

Reply:
not for this. we have normalizer functions defined in leave list